Seperate the GLib code from GtkContext.

eafton opened the pull request:
I want make IsTraySupported check if you have a tray extension installed on GNOME desktops, I have to use GSettings for that, so I want to split off the GLib code.

mikela-valve commented:
Hi @eafton, can you further describe the proposed change to SDL_tray and why you would prefer having a separate initialization for glib-only rather than initializing both glib and gtk then making the GSettings calls necessary?

Additionally we'd prefer to keep the glib/gtk parts of SDL_gtk distinct and so things like copied typedefs were intentionally left with `SDL_` prefixes (e.g. `guint` instead of `SDL_guint`). Similarly with the `gtk` struct in `SDL_GtkContext` which was added to stand in for the `gtk_` prefix that was removed from the function pointer names.

eafton commented:
> Hi @eafton, can you further describe the proposed change to SDL_tray and why you would prefer having a separate initialization for glib-only rather than initializing both glib and gtk then making the GSettings calls necessary?
> 
> Additionally we'd prefer to keep the glib/gtk parts of SDL_gtk distinct and so things like copied typedefs were intentionally left with `SDL_` prefixes (e.g. `guint` instead of `SDL_guint`). Similarly with the `gtk` struct in `SDL_GtkContext` which was added to stand in for the `gtk_` prefix that was removed from the function pointer names.

I want to add a check in SDL_IsTraySupported that checks if a tray shell extension is installed when running under GNOME, and return false/true accordingly. I need to use GSettings from GLib for checking what shell extensions are installed (this can all be bypassed with a hint).

I can reverse adding the SDL_ prefix on to type names when I get home and have access to my laptop, I added them because other internal type names like the ones in the X11 messagebox code have them, and an option to link against GLib might be added in the future (SDL links to libdecor which in turn brings in GLib through Pango/GTK)

I removed the gtk struct from gtkcontext because the gtk context variable is already named gtk in the places where its used.
